autoload/lsp/completion.vim

```diff
@@ -228,8 +228,9 @@ export def CompletionReply(lspserver: dict<any>, cItems: any)
       # snippet completion.  Needs a snippet plugin to expand the snippet.
       # Remove all the snippet placeholders
       d.word = MakeValidWord(d.word)
-    elseif !lspserver.completeItemsIsIncomplete
-      # Don't attempt to filter on the items, when "isIncomplete" is set
+    elseif !lspserver.completeItemsIsIncomplete || opt.lspOptions.useBufferCompletion
+      # Filter items only when "isIncomplete" is set (otherwise server would
+      #   have done the filtering) or when buffer completion is enabled
```

Comment:
|| opt.lspOptions.useBufferCompletion ?

i felt i have no confident on this, or if this needed, then how about those snip compl etc options?

// bufcompl had been noticed that's risk (or perf impact) option which had noticed in the doc, so i think user should have awareness be careful to enable it.